internal/manager/run.go

```diff
@@ -64,15 +65,30 @@ func Run(ctx context.Context, c *Config, diagnostic util.ConfigDumpDiagnostic) e
 	}
 
 	setupLog.Info("getting the kong admin api client configuration")
-	kongConfig, err := setupKongConfig(ctx, setupLog, c)
+	adminClient, err := c.GetKongClient(ctx)
 	if err != nil {
-		return fmt.Errorf("unable to build the kong admin api configuration: %w", err)
-	}
+		return fmt.Errorf("unable to build kong api client: %w", err)
+	}
+
+	var kongRoot map[string]interface{}
+	err = retry.Do(
+		func() error {
+			kongRoot, err = adminClient.Root(ctx)
+			return err
+		},
+		retry.Attempts(c.KongAdminAPIRetries),
+		retry.DelayType(retry.FixedDelay),
+		retry.OnRetry(func(n uint, err error) {
+			setupLog.Info("Retrying kong admin api client call #%d/%d after error: %w", n, c.KongAdminAPIRetries, err)
+		}),
+	)
 
-	kongRoot, err := kongConfig.Client.Root(ctx)
 	if err != nil {
 		return fmt.Errorf("could not retrieve Kong admin root: %w", err)
 	}
+
+	kongConfig := setupKongConfig(ctx, adminClient, setupLog, c)
```

Comment:
Changed up the order of function calls here. This is because we should ensure that Kong Admin API connection is established before making any API calls. 

Right now, we are calling the Tags Service API before calling the Root API call. This leads to a misleading error message to users like
```
time="2022-02-17T09:51:28Z" level=error msg="tag filtering disabled because Kong Admin API does not support tags" error="making HTTP request: Get \"https://localhost:8444/tags\": dial tcp 127.0.0.1:8444: connect: connection refused" logger=setup
2
```